packages/lodestar/src/chain/chain.ts

```diff
@@ -93,6 +93,9 @@ export class BeaconChain extends EventEmitter implements IBeaconChain {
     await this.runStateTransition(block, this.latestState);
     await this.opPool.processBlockOperations(block);
 
+    // check for duplicate proposal
+    await this.opPool.checkDuplicateProposer(this.config, block);
```

Comment:
opPool already receives every processed block in `processBlockOperations` method, so you should invoke checking for slashings there

Reply:
Without access to the `config` object it makes things extremely difficult, could i pass it down from the `BeaconNode` class?

Reply:
nvm i passed it in -llooks clean